include/eve/module/real/core/function/regular/simd/x86/abs.hpp

```diff
@@ -17,38 +17,85 @@
 namespace eve::detail
 {
   template<real_scalar_value T, typename N, x86_abi ABI>
-  EVE_FORCEINLINE wide<T, N, ABI> abs_(EVE_SUPPORTS(ssse3_), wide<T, N, ABI> const &v) noexcept
+  EVE_FORCEINLINE wide<T, N, ABI> abs_(EVE_SUPPORTS(sse2_), wide<T, N, ABI> const &v) noexcept
   {
-    constexpr auto cat = categorize<wide<T, N, ABI>>();
+    constexpr auto c = categorize<wide<T, N, ABI>>();
 
-          if constexpr( cat && category::unsigned_) return v;
-    else  if constexpr( cat && category::float_   ) return bit_notand(mzero(eve::as(v)), v);
-    else  if constexpr( cat && category::size64_  ) return map(eve::abs, v);
-    else  if constexpr( cat == category::int32x4  ) return _mm_abs_epi32(v);
-    else  if constexpr( cat == category::int16x8  ) return _mm_abs_epi16(v);
-    else  if constexpr( cat == category::int8x16  ) return _mm_abs_epi8(v);
-    else  if constexpr(current_api >= avx2)
-    {
-          if constexpr( cat == category::int32x8  ) return _mm256_abs_epi32(v);
-    else  if constexpr( cat == category::int16x16 ) return _mm256_abs_epi16(v);
-    else  if constexpr( cat == category::int8x32  ) return _mm256_abs_epi8(v);
-    }
-    else                                            return aggregate(eve::abs, v);
+          if constexpr(c && category::unsigned_  )  return v;
+    else  if constexpr(c == category::float32x16 )  return _mm512_abs_ps(v);
+    else  if constexpr(c == category::float64x8  )  return _mm512_abs_pd(v);
+    else  if constexpr(c && category::float_     )  return bit_notand(mzero(as(v)), v);
```

Comment:
What's up with zeroes at the beginning?

Reply:
mzero is 0x80000000 so notand with a float sets the bit of sign to zero in a single instruction